pkg/models/terminal/terminal.go

```diff
@@ -224,3 +355,49 @@ func (t *terminaler) HandleSession(shell, namespace, podName, containerName stri
 
 	session.Close(1, "Process exited")
 }
+
+func (t *terminaler) HandleShellAccessToNode(nodename string, conn *websocket.Conn) {
+	ch := make(chan bool, 1)
+	once.Do(func() {
+		NodeSessionCounter = make(map[string]*int64)
+	})
+
+	nodeTerminaler, err := NewNodeTerminaler(nodename, t.options, t.client)
+	if err != nil {
+		klog.Warning("node terminaler init error: ", err)
+		return
+	}
+
+	pod, err := nodeTerminaler.getNSEnterPod()
+	if err != nil {
+		klog.Warning("get nsenter pod error: ", err)
+		return
+	}
+
+	go nodeTerminaler.WatchPodStatusBeRunning(pod, ch)
+	select {
+	case flag := <-ch:
+		if flag {
+			t.HandleSession(nodeTerminaler.Shell, nodeTerminaler.Namespace, nodeTerminaler.PodName, nodeTerminaler.ContainerName, conn)
+			defer nodeTerminaler.CleanUpNSEnterPod()
+		}
+	}
+}
+
+func (n *NodeTerminaler) WatchPodStatusBeRunning(pod *v1.Pod, ch chan bool) {
+	var err error
+	for true {
+		if pod.Status.Phase == v1.PodRunning {
+			atomic.AddInt64(NodeSessionCounter[n.Nodename], 1)
+			ch <- true
+			break
+		}
+
+		pod, err = n.client.CoreV1().Pods(pod.ObjectMeta.Namespace).Get(context.Background(), pod.ObjectMeta.Name, metav1.GetOptions{})
+		if err != nil {
+			klog.Warning(err)
+			ch <- false
```

Comment:
You can use close(ch) to close the unbuffered channel and notify the main process. Since this practice is common.

Reply:
if I use close(ch) here, I cannot perform error detection.

Reply:
OK. Adding two unbuffered channels is another way to solve this case.

Reply:
OK.I'll modify it

Reply:
> OK.I'll modify it

No need to change. If it can work properly, let it go.